guides/plugins/plugins/framework/custom-field/fetching-data-from-entity-selection.md

```diff
@@ -0,0 +1,237 @@
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Subscriber;
+
+use Shopware\Core\Content\Product\ProductEntity;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Event\EntityLoadedEvent;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Shopware\Core\Content\Product\ProductEvents;
+
+class ProductSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $productRepository;
+
+    public function __construct(
+        EntityRepositoryInterface $productRepository
+    ) {
+        $this->productRepository = $productRepository;
+    }
+
+   //...
+}
+```
+{% endcode %}
+
+As you can see, the product repository was injected and is now available to us.
+The last step is to resolve the `custom_lined_product` value inside the `onProductLoaded` method.
+
+Here is now the final subscriber implementation.
+
+{% code title="<plugin root>/src/Subscriber/ProductSubscriber.php" %}
+```php
+<?php declare(strict_types=1);
+
+namespace Swag\BasicExample\Subscriber;
+
+use Shopware\Core\Content\Product\ProductEntity;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Event\EntityLoadedEvent;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Shopware\Core\Content\Product\ProductEvents;
+
+class ProductSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $productRepository;
+
+    public function __construct(
+        EntityRepositoryInterface $productRepository
+    ) {
+        $this->productRepository = $productRepository;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ProductEvents::PRODUCT_LOADED_EVENT => 'onProductLoaded'
+        ];
+    }
+
+    public function onProductLoaded(EntityLoadedEvent $event): void
+    {
+        /** @var ProductEntity $productEntity */
+        foreach ($event->getEntities() as $productEntity) {
+            $customFields = $productEntity->getCustomFields();
+
+            foreach($customFields as $name => $value) {
+                if ($name !== 'custom_demo_test' || emtpy($value)) {
+                    return;
+                }
+
+                $context = $event->getContext();
+
+                /** @var ProductEntity $productEntity */
+                $product = $this->productRepository
+                    ->search(new Criteria([$value]), $context)->first();
+                
+                $customFields[$name] = $product;
+            }
+
+            $productEntity->setCustomFields($customFields);
+        }
+    }
+}
+```
+{% endcode %}
+
+
+## Next steps
+...
```

Comment:
Should this be updated ?

Reply:
removed next steps - couldn't find any useful things to mention here